src/psyclone/transformations.py

```diff
@@ -250,11 +273,11 @@ def apply(self, node1, node2):
         keep = Memento(schedule, self, [node1, node2])
 
         # add loop contents of node2 to node1
-        node1.children.extend(node2.children)
+        node1.loop_body._children.extend(node2.loop_body)
 
         # change the parent of the loop contents of node2 to node1
-        for child in node2.children:
-            child.parent = node1
+        for child in node2.loop_body:
+            child.parent = node1.children[3]
```

Comment:
Could `children[3]` be `loop_body`? i.e. does `loop_body` return the `Schedule` of the loop body?

Reply:
Done